src/components/CompareResults/CompareResultsTableHead.tsx

```diff
@@ -0,0 +1,171 @@
+import FilterOptionsList from './FilterOptionsList';
+
+const tableHead: CompareResultsTableHeader[] = [
+  {
+    id: 'platform',
+    label: 'Platform',
+    key: 'platform',
+    align: 'center',
+  },
+  {
+    id: 'graph',
+    label: 'Graph',
+    key: 'graph',
+    align: 'center',
+  },
+  {
+    id: 'test-name',
+    label: 'Test Name',
+    key: 'test',
+    align: 'left',
+  },
+  {
+    id: 'base-value',
+    label: 'Base',
+    key: 'base',
+    align: 'center',
+  },
+  {
+    id: 'new-value',
+    label: 'New',
+    key: 'new',
+    align: 'center',
+  },
+  {
+    id: 'delta-percent',
+    label: 'Delta',
+    key: 'delta',
+    align: 'center',
+  },
+  {
+    id: 'confidence',
+    label: 'Confidence',
+    key: 'confidence',
+    align: 'center',
+  },
+  {
+    id: 'total-runs',
+    label: 'Total Runs',
+    key: 'run',
+    align: 'center',
+  },
+];
+
+interface FilterOptions {
+  platform: Set<Platform>;
+  test: Set<string>;
+  confidence: Set<ConfidenceText>;
+}
+
+const filterOptions: FilterOptions = {
+  platform: new Set([]),
+  test: new Set([]),
+  confidence: new Set(['low', 'med', 'high'] as ConfidenceText[]),
+};
+
+const filterKeys = Object.keys(filterOptions);
+
+const CompareResultsTableHead = () => {
+  const compareResults: CompareResultsItem[] = useAppSelector(
+    (state) => state.compareResults.data,
+  );
+
+  filterOptions.platform = new Set(
+    compareResults.map((result) => result.platform),
+  );
+
+  filterOptions.test = new Set([
+    ...compareResults.map((result) => result.test),
+  ]);
+
+  const [openFilter, setOpenFilter] = React.useState<string | null>(null);
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(
+    null,
+  );
+
+  const open = Boolean(anchorEl);
+
+  const handleOpenOptions = (
+    event: React.MouseEvent<HTMLButtonElement>,
+    key: string,
+  ) => {
+    setOpenFilter(key);
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleCloseOptions = () => {
+    setAnchorEl(null);
+    setOpenFilter(null);
+  };
+
+  return (
+    <TableHead>
+      <TableRow>
+        {tableHead.map(
+          ({ label, key, align }: CompareResultsTableHeader, index) => {
+            let options: string[] | ConfidenceText[] = [];
+            let id = '';
+            const headerId = key as keyof typeof filterOptions;
+            if (filterKeys.includes(headerId)) {
+              options = Array.from(filterOptions[headerId].values());
+              id = open ? `${key as keyof typeof filterOptions}` : '';
+            }
+
+            return (
+              <TableCell key={index} align={align}>
+                {label}
+                {filterKeys.includes(headerId) && (
+                  <React.Fragment>
+                    <IconButton
+                      data-testid={`${headerId}-options-button`}
+                      onClick={(e) => handleOpenOptions(e, headerId)}
+                    >
+                      <MoreVertIcon />
```

Comment:
I'm happy to defer to your judgement or user feedback, but my thought are that an icon like `FilterList` or `Tune` (https://mui.com/material-ui/material-icons/?query=filter&selected=FilterList) might better convey its purpose? Thoughts?

also could you file an issue for setting column widths so that the icon is always next to, rather than below the header?

Reply:
I agree with the `FilterList` icon.

[Here is the issue ticket for the width](https://mozilla-hub.atlassian.net/browse/FXP-2471)